### PR TITLE
feat: provide declaration kind and provide JS interface to private

### PR DESCRIPTION
**Pull request by kitsonk**
This PR does two things:

- Adds a `declaration_type` property to a `DocNode` which is set to the _reason_ why the doc node exists.
- Allowed the JavaScript/TypeScript interface to use the _private_ setting which includes all doc nodes/symbols in the output, not just those being exported.

This is designed to make it possible to generate documentation for everything in a module or ambient type declaration file and display that information usefully. This will allow us to build towards symbol references and scope analysis.

### CODE REVIEW

File: lib/types.d.ts [hunk 1]
```diff
@@ -11,10 +11,13 @@ export type DocNode =
   | DocNodeInterface
   | DocNodeImport;
 
+export type DeclarationKind = "private" | "export" | "import" | "declaration";
```
Comment (lucacasonato) [inline lib/types.d.ts:14]: Having a comment here explaining what each of these mean would be very useful. Does `"declaration"` mean the `declare` keyword in TS?

**Review by lucacasonato — APPROVED** [review]

File: lib/types.d.ts (hunk 1 above)
Comment (kitsonk) [reply lib/types.d.ts:14]: Hmmm... yeah, maybe `declare` is better, yes that is what it means.  I will change it to `declare` and add some comments.

**Comment by lucacasonato** [comment]
Actually, one more comment:

Would it not make sense to remove the `import` kind and just mark imports as a `private` declaration? The distinction between `import` and `private` is not clear to me, because imports can already be found by looking for nodes with `kind: "import"`.

**Comment by kitsonk** [comment]
That makes sense, will amend.

**Pull request merged**